In [2]:
import pandas as pd
import numpy as np
#import matplotlib.pyplot as plt
#import seaborn as sns

In [3]:
df = pd.read_csv('total_data(2016제외) (1).csv')

### xdata로 모든 column을 넣어서 Ridge
- Ridge regression으로 높은 weight를 가지는 feature를 선택

In [4]:
# 필요 패키지 로드
#from sklearn.preprocessing import MinMaxScaler
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import Ridge
from sklearn.metrics import mean_absolute_error

In [5]:
# y값인 q1-q5가 결측인 2020년 데이터 제거
a = df[0:-82]
a

jur_stn       year  report_sx_남성  report_sx_불상  report_sx_여성  \
0     서울중부경찰서  2017(상반기)       13404.0        3239.0        5931.0   
1     서울종로경찰서  2017(상반기)       11579.0        2560.0        5534.0   
2    서울남대문경찰서  2017(상반기)        9093.0        1855.0        3551.0   
3    서울서대문경찰서  2017(상반기)       18441.0       10029.0       10298.0   
4     서울혜화경찰서  2017(상반기)       10829.0        2989.0        5038.0   
..        ...        ...           ...           ...           ...   
241   창원서부경찰서  2019(하반기)       12630.0        2990.0        7151.0   
242   마산중부경찰서  2019(하반기)       12223.0        2552.0        6489.0   
243   마산동부경찰서  2019(하반기)       15177.0        3190.0        7647.0   
244     진주경찰서  2019(하반기)       21840.0        5262.0       11455.0   
245     진해경찰서  2019(하반기)       11600.0        2442.0        6278.0   

     report_sx_total  vio_cnt  mur_rob_cnt  car_cnt  5m_crm_yn  ...  \
0            22574.0   1324.0          0.0   1412.0     3275.0  ...   
1            19674.0   1102.0          0.0    867.0     2914.0  ...   
2            14499.0    949.0          0.0    736.0     2506.0  ...   
3            38769.0   1924.0          0.0   1788.0     4267.0  ...   
4            18857.0   1127.0          0.0    899.0     2864.0  ...   
..               ...      ...          ...      ...        ...  ...   
241          22771.0   1460.0          2.0   1964.0     2142.0  ...   
242          21264.0   1265.0          5.0   1584.0     1369.0  ...   
243          26014.0   1715.0          8.0   1910.0     1920.0  ...   
244          38557.0   2552.0          0.0   3570.0     2876.0  ...   
245          20320.0   1444.0          0.0   1692.0     1354.0  ...   

     single_f_2030  single_f_4050  single_f_o60  adult  bell    q1    q2  \
0           2128.0         1230.5        1635.5    0.0   0.0  75.2  76.5   
1           2703.0         1378.5        1782.5    2.0   2.0  81.6  82.4   
2           2128.0         1230.5        1635.5    0.0   0.0  77.3  79.3   
3          12420.0         4496.0        7066.0    0.0   0.0  78.0  80.0   
4           2703.0         1378.5        1782.5    0.0   0.0  78.6  80.6   
..             ...            ...           ...    ...   ...   ...   ...   
241         2900.0         4492.0        5574.0    1.0   1.0  79.1  79.5   
242         2216.0         2837.0        7186.0    3.0   3.0  80.8  80.0   
243         1496.0         2939.0        6166.0    1.0   1.0  75.0  76.5   
244         6026.0         5205.0       11823.0    8.0   8.0  78.7  79.6   
245         1782.0         2592.0        4931.0    3.0   3.0  76.6  79.2   

       q3    q4    q5  
0    69.9  65.9  72.6  
1    69.1  62.7  72.2  
2    68.8  61.0  71.7  
3    67.0  66.3  71.4  
4    70.7  65.2  75.4  
..    ...   ...   ...  
241  68.5  71.2  76.0  
242  70.8  73.2  76.5  
243  69.4  68.9  73.1  
244  70.3  74.6  75.6  
245  67.0  70.9  73.5  

[246 rows x 75 columns]

In [6]:
# 경찰서와 연도 데이터 제거
a.drop(columns = ['jur_stn', 'year'], inplace = True)

C:\Anaconda3\lib\site-packages\pandas\core\frame.py:4906: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().drop(


In [7]:
a_ = a.drop(columns = ['q1', 'q2', 'q3', 'q4', 'q5'])

In [8]:
# StandardScaling
scaler = StandardScaler()
scaler.fit(a_) # scaler에 xdata 학습
a_s = scaler.transform(a_)
a_s

array([[-0.87369027, -0.8125045 , -1.08891897, ..., -1.55004946,
        -0.59342342, -0.59342342],
       [-1.07373936, -1.09735738, -1.17135971, ..., -1.50210803,
         0.29943384,  0.29943384],
       [-1.34624459, -1.39311773, -1.58314811, ..., -1.55004946,
        -0.59342342, -0.59342342],
       ...,
       [-0.67934122, -0.83306089, -0.7325756 , ..., -0.07250741,
        -0.14699479, -0.14699479],
       [ 0.05102978,  0.03618089,  0.05819103, ...,  1.77242259,
         2.97800563,  2.97800563],
       [-1.07143743, -1.14686053, -1.01686119, ..., -0.47528071,
         0.74586247,  0.74586247]])

In [9]:
# 2017, 2018 데이터를 학습용, 2019년 데이터를 검증용 데이터셋으로 설정
train_ = a_s[:-82]
test_ = a_s[-82:]
train = a[:-82]
test = a[-82:]

In [10]:
xtrain = train_
ytrain1 = train['q1']
ytrain2 = train['q2']
ytrain3 = train['q3']
ytrain4 = train['q4']
ytrain5 = train['q5']

In [11]:
xtest = test_
ytest1 = test['q1']
ytest2 = test['q2']
ytest3 = test['q3']
ytest4 = test['q4']
ytest5 = test['q5']

In [12]:
a_.columns

Index(['report_sx_남성', 'report_sx_불상', 'report_sx_여성', 'report_sx_total',
       'vio_cnt', 'mur_rob_cnt', 'car_cnt', '5m_crm_yn', 'theft_violence_x',
       'robber_murder_x', 'traffic_x', 'public_x', 'crm', 'theft_violence_y',
       'robber_murder_y', 'traffic_y', 'public_y', 'light_cnt', '차대사람', '차대차',
       '차량단독', '가해_성별_기타불명', '가해_남', '가해_여', '가해_10대이하', '가해_20대', '가해_30대',
       '가해_40대', '가해_50대', '가해_60대', '가해_70대', '가해_80대이상', '가해_미분류', '피해_남',
       '피해_미분류', '피해_여', '피해_10대이하', '피해_20대', '피해_30대', '피해_40대', '피해_50대',
       '피해_60대', '피해_70대', '피해_80대이상', 'cctv', 'pol_num', 'popu_합계', 'for_합계',
       'popu_u20', 'popu_2030', 'popu_4050', 'popu_o60', 'for_u20', 'for_2030',
       'for_4050', 'for_o60', 'single_합계', 'single_u20', 'single_2030',
       'single_4050', 'single_o60', 'single_f_합계', 'single_f_u20',
       'single_f_2030', 'single_f_4050', 'single_f_o60', 'adult', 'bell'],
      dtype='object')

In [13]:
# 그리드 서치 패키지
from sklearn.model_selection import GridSearchCV

In [14]:
# 최적 성능을 내는 lasso의 alpha값을 얻기 위해 param_grid 생성

num = []
for i in range(1, 10000):
    num.append(i * 0.001)

param_grid = {'alpha' : num}

#그리드 서치 설정
#grid_search = GridSearchCV(Ridge(), param_grid = param_grid, cv = 10, n_jobs = -1, scoring ='r2')
grid_search = GridSearchCV(Ridge(), param_grid = param_grid, cv = 10, n_jobs = -1, scoring ='neg_mean_absolute_error')

#### q1 절도폭력 

In [15]:
# 그리드 서치 후 최고 성능의 모델을 lasso1에 저장

grid_search.fit(xtrain, ytrain1)
ridge1 = grid_search.best_estimator_

In [16]:
# MAE 출력

y_pred1 = ridge1.predict(xtest)
mean_absolute_error(ytest1, y_pred1)

2.4316172928573057

In [17]:
# 결과

print('alpha =', ridge1.alpha)
print(ridge1.coef_) # Ridge 회귀분석으로 나온 weghit값

print('가장 강한 양의 상관관계: ',a_.columns[ridge1.coef_.argmax()], '\n가장 강한 음의 상관관계: ', a_.columns[ridge1.coef_.argmin()])

alpha = 8.952
[-0.33483957  0.37661366 -0.48523996 -0.28064439 -0.8990635   0.15751145
  0.41291469  0.13687925  0.11960418  0.35688615  0.22220958 -0.06252449
 -0.36854802 -0.2445179  -0.1413476   0.25413814  0.5172975   0.4593124
  0.16701212 -0.08332483 -0.10135646  0.18843532 -0.19234326  0.58345782
 -0.85889268 -0.76952162  0.1308204   0.54639266 -0.30694238  0.10580306
  0.62551964 -0.05664475  0.12403315 -0.27938853 -0.24203103  0.71781408
 -0.38129038 -0.14724786  0.61895816 -0.17047703 -0.40959815 -0.23254253
  0.75640527 -0.41602122  0.07375925  0.42169254  0.28458947 -0.41028199
  1.04095646  0.01570212  0.03345439  0.40501422  0.47496167 -0.83882328
 -0.27116514 -0.21946178 -0.19888495  0.5330361   0.14526975 -0.82865693
 -0.49110295  0.25996894 -0.17966631  0.49328196 -0.09900065 -0.07097933
 -0.04335506 -0.04335506]
가장 강한 양의 상관관계:  popu_u20 
가장 강한 음의 상관관계:  vio_cnt


In [18]:
# q1 관련 feature
print('상위 30개 feature\n', a_.columns[sorted(range(len(ridge1.coef_)), key=lambda i: abs(ridge1.coef_[i]))[-30:]])

상위 30개 feature
 Index(['report_sx_남성', 'robber_murder_x', 'crm', 'report_sx_불상', '피해_10대이하',
       'popu_o60', '피해_50대', 'for_합계', 'car_cnt', '피해_80대이상', 'pol_num',
       'light_cnt', 'for_u20', 'report_sx_여성', 'single_o60', 'single_f_2030',
       'public_y', 'single_u20', '가해_40대', '가해_여', '피해_30대', '가해_70대', '피해_여',
       '피해_70대', '가해_20대', 'single_4050', 'for_2030', '가해_10대이하', 'vio_cnt',
       'popu_u20'],
      dtype='object')


#### q2 강도살인

In [19]:
# 그리드 서치 후 최고 성능의 모델을 lasso2에 저장

grid_search.fit(xtrain, ytrain2)
ridge2 = grid_search.best_estimator_

In [20]:
# MAE 출력

y_pred2 = ridge2.predict(xtest)
mean_absolute_error(ytest2, y_pred2)

2.382736083231455

In [21]:
# 결과

print('alpha =', ridge2.alpha)
print(ridge2.coef_) # Ridge 회귀분석으로 나온 weghit값

print('가장 강한 양의 상관관계: ',a_.columns[ridge2.coef_.argmax()], '\n가장 강한 음의 상관관계: ', a_.columns[ridge2.coef_.argmin()])

alpha = 2.104
[-0.25860875 -0.04911503 -0.59300995 -0.33306738 -1.59845421 -0.02122718
  1.27609744  0.01972126  0.07221074  0.24870911  0.54725118  0.03450694
 -1.04641601 -0.11601711  0.02666477  0.39650661  0.53096802  0.42899395
  0.85712413 -0.45801871  0.37736323  0.43229552 -0.48839705  1.09443192
 -0.84243183 -1.09406365 -0.56529752  1.25983758 -0.39966024 -0.29651648
  1.04386208 -0.1498086   0.1000953  -0.25842585 -0.38316871  0.17437833
  0.14284252  0.39290754  0.89503135 -1.08973028 -1.02858497 -0.145086
  1.2927875  -0.87989586  0.07248647  0.59494174  0.36692909 -0.44177674
  0.86821059 -0.06922947 -0.22299389  1.45700444  1.28188454 -1.15924696
 -0.22015899 -0.25903431 -0.16009612  1.1669907   0.40497438 -1.1250026
 -0.80515392  0.32964882 -0.9719177   0.5558799  -0.01219731  0.03050821
  0.0173043   0.0173043 ]
가장 강한 양의 상관관계:  popu_o60 
가장 강한 음의 상관관계:  vio_cnt


In [22]:
# q2 관련 feature
print('상위 30개 feature\n', a_.columns[sorted(range(len(ridge2.coef_)), key=lambda i: abs(ridge2.coef_[i]))[-30:]])

상위 30개 feature
 Index(['차대차', '가해_남', 'public_y', 'traffic_x', 'single_f_2030', '가해_30대',
       'report_sx_여성', 'pol_num', 'single_o60', '가해_10대이하', '차대사람', 'popu_u20',
       '피해_80대이상', '피해_30대', 'single_f_u20', '피해_50대', '가해_70대', 'crm',
       '피해_40대', '가해_20대', '가해_여', 'single_4050', 'for_2030', 'single_u20',
       '가해_40대', 'car_cnt', 'for_u20', '피해_70대', 'popu_o60', 'vio_cnt'],
      dtype='object')


#### q3 교통안전

In [23]:
# 그리드 서치 후 최고 성능의 모델을 lasso3에 저장

grid_search.fit(xtrain, ytrain3)
ridge3 = grid_search.best_estimator_

In [24]:
# MAE 출력

y_pred3 = ridge3.predict(xtest)
mean_absolute_error(ytest3, y_pred3)

2.274900251538899

In [25]:
# 결과

print('alpha =', ridge3.alpha)
print(ridge3.coef_) # Ridge 회귀분석으로 나온 weghit값

print('가장 강한 양의 상관관계: ',a_.columns[ridge3.coef_.argmax()], '\n가장 강한 음의 상관관계: ', a_.columns[ridge3.coef_.argmin()])

alpha = 9.999
[-6.86677816e-02 -9.28757018e-03  1.57217219e-01  6.85913846e-03
 -1.26408071e-01  1.91304210e-01  6.12038511e-02 -9.06044415e-02
 -6.00871327e-03  2.58327814e-01 -4.34289375e-01 -3.71848654e-02
  2.62592098e-01 -2.55559105e-05 -1.82990937e-01 -2.78896770e-01
  2.36899408e-01  2.28885088e-01 -3.48180061e-01  6.25536888e-02
 -5.19526037e-02  8.84042648e-02 -1.34806565e-01  3.71579011e-01
 -9.39020334e-01 -4.66816071e-01 -1.71786883e-01  3.86187411e-01
  8.07861329e-02 -7.72586708e-02  3.75976015e-01  1.97509895e-01
  1.82962277e-01 -5.82817748e-02 -5.55733732e-02  7.07653670e-02
  5.30555567e-01 -3.23874691e-01 -2.86643664e-02 -1.85695267e-01
  2.05824625e-01 -1.78984349e-01  8.54404589e-01  1.06189378e-01
  2.84501548e-01  6.12196808e-01  1.48735260e-01 -1.41193354e-01
  1.42137178e-01  1.54811862e-02  2.74646232e-01  1.45729254e-01
  4.48925249e-01 -4.45969542e-01 -4.00301159e-02 -4.39544954e-02
 -2.19333487e-01  2.55974449e-02  1.71170393e-02 -6.56594974e-01
 -3.5237390

In [26]:
# q3 관련 feature
print('상위 30개 feature\n', a_.columns[sorted(range(len(ridge3.coef_)), key=lambda i: abs(ridge3.coef_[i]))[-30:]])

상위 30개 feature
 Index(['robber_murder_y', '피해_40대', 'mur_rob_cnt', '가해_80대이상', '피해_50대',
       'single_합계', 'light_cnt', 'public_y', 'robber_murder_x', 'crm',
       'popu_4050', 'traffic_y', 'cctv', '피해_20대', 'single_f_4050', '차대사람',
       'single_o60', 'single_f_2030', '가해_여', '가해_70대', '가해_40대', 'traffic_x',
       'for_2030', 'for_u20', '가해_20대', '피해_10대이하', 'pol_num', 'single_4050',
       '피해_70대', '가해_10대이하'],
      dtype='object')


#### q4 법질서 준수도

In [27]:
# 그리드 서치 후 최고 성능의 모델을 lasso4에 저장

grid_search.fit(xtrain, ytrain4)
ridge4 = grid_search.best_estimator_

In [28]:
# MAE 출력

y_pred4 = ridge4.predict(xtest)
mean_absolute_error(ytest4, y_pred4)

2.957983394039355

In [29]:
# 결과

print('alpha =', ridge4.alpha)
print(ridge4.coef_) # Ridge 회귀분석으로 나온 weghit값

print('가장 강한 양의 상관관계: ',a_.columns[ridge4.coef_.argmax()], '\n가장 강한 음의 상관관계: ', a_.columns[ridge4.coef_.argmin()])

alpha = 5.4110000000000005
[-0.28732503  0.3529043  -0.03324029 -0.12139755 -0.6620332   1.07781985
  1.47781156 -0.72155308 -0.70568114  0.45094694  0.10549896 -0.06787264
 -0.53328532 -0.47713235 -0.18315066  0.81411381 -0.09058703  0.36922156
 -0.08794376  0.08025125 -0.15349688  0.4939797  -0.05719471  0.35083668
 -1.20719465 -1.1581795   0.32895653  1.16700667 -0.34680852 -0.14439415
  1.1053799  -0.11791399  0.38272839  0.04364713 -0.77132055  0.1402522
  0.98418383  0.47401715 -0.20261852  0.08242802 -0.74081728  0.34927619
  1.60255405 -0.63671274  0.7058158  -1.67741546  0.19713299 -0.28151045
  0.95291919 -0.47334949  0.20823421  0.57323176  0.82834007 -0.4502488
 -0.35527091 -0.13976825 -0.22292614 -0.10759479  0.14657508 -1.22214932
 -0.15125463  0.57255033 -0.10842542  1.0080193  -0.5355187   0.17510068
 -0.13805612 -0.13805612]
가장 강한 양의 상관관계:  피해_70대 
가장 강한 음의 상관관계:  pol_num


In [30]:
# q3 관련 feature
print('상위 30개 feature\n', a_.columns[sorted(range(len(ridge4.coef_)), key=lambda i: abs(ridge4.coef_[i]))[-30:]])

상위 30개 feature
 Index(['robber_murder_x', 'popu_2030', '피해_20대', 'theft_violence_y',
       '가해_성별_기타불명', 'crm', 'single_f_4050', 'single_f_합계', 'popu_o60',
       '피해_80대이상', 'vio_cnt', 'theft_violence_x', 'cctv', '5m_crm_yn',
       '피해_50대', '피해_미분류', 'traffic_y', 'for_u20', 'popu_u20', '피해_10대이하',
       'single_f_2030', 'mur_rob_cnt', '가해_70대', '가해_20대', '가해_40대',
       '가해_10대이하', 'single_4050', 'car_cnt', '피해_70대', 'pol_num'],
      dtype='object')


#### q5 전반적 안전도

In [31]:
# 그리드 서치 후 최고 성능의 모델을 lasso4에 저장

grid_search.fit(xtrain, ytrain5)
ridge5 = grid_search.best_estimator_

In [32]:
# MAE 출력

y_pred5 = ridge5.predict(xtest)
mean_absolute_error(ytest5, y_pred5)

2.2458133802946314

In [33]:
# 결과

print('alpha =', ridge5.alpha)
print(ridge5.coef_) # Ridge 회귀분석으로 나온 weghit값

print('가장 강한 양의 상관관계: ',a_.columns[ridge5.coef_.argmax()], '\n가장 강한 음의 상관관계: ', a_.columns[ridge5.coef_.argmin()])

alpha = 5.517
[-0.31403888  0.33105522 -0.25742705 -0.20693952 -1.02612135  0.2699458
  0.8377137  -0.20378789 -0.01997183  0.34165809  0.25872943  0.0492724
 -0.4063197  -0.52137071 -0.03889006  0.36055554  0.17986035  0.4662497
  0.20347241 -0.06344053 -0.2318441   0.33782374 -0.11414883  0.34342262
 -0.65480894 -0.79289291  0.05026206  0.68699034 -0.14987391 -0.14425473
  0.70314852  0.04445099  0.17372055 -0.07967947 -0.45501355  0.24878072
  0.14444119  0.11689535  0.26930828 -0.27242424 -0.37852907  0.00718668
  1.12202846 -0.40925522  0.24051634  0.05918084  0.23415176 -0.33540509
  0.91532193 -0.17368781  0.12122418  0.43757023  0.54976682 -0.71525246
 -0.24369458 -0.1371852  -0.24333848  0.40026122  0.08279732 -0.89493629
 -0.43270885  0.32532794 -0.31944155  0.62313296 -0.12087053 -0.09585451
 -0.01595209 -0.01595209]
가장 강한 양의 상관관계:  피해_70대 
가장 강한 음의 상관관계:  vio_cnt


In [34]:
# q3 관련 feature
print('상위 30개 feature\n', a_.columns[sorted(range(len(ridge5.coef_)), key=lambda i: abs(ridge5.coef_[i]))[-30:]])

상위 30개 feature
 Index(['report_sx_남성', 'single_f_u20', 'single_f_합계', 'report_sx_불상', 'for_합계',
       '가해_성별_기타불명', 'robber_murder_x', '가해_여', 'traffic_y', '피해_50대',
       'single_u20', 'crm', '피해_80대이상', 'single_o60', 'popu_o60', '피해_미분류',
       'light_cnt', 'theft_violence_y', 'for_u20', 'single_f_2030', '가해_10대이하',
       '가해_40대', '가해_70대', 'for_2030', '가해_20대', 'car_cnt', 'single_4050',
       'popu_u20', 'vio_cnt', '피해_70대'],
      dtype='object')


### Ridge Regression 결과

- q1  
    - alpha = 8.952  
    - 상위 30개  
        ['report_sx_남성', 'robber_murder_x', 'crm', 'report_sx_불상', '피해_10대이하',
       'popu_o60', '피해_50대', 'for_합계', 'car_cnt', '피해_80대이상', 'pol_num',
       'light_cnt', 'for_u20', 'report_sx_여성', 'single_o60', 'single_f_2030',
       'public_y', 'single_u20', '가해_40대', '가해_여', '피해_30대', '가해_70대', '피해_여',
       '피해_70대', '가해_20대', 'single_4050', 'for_2030', '가해_10대이하', 'vio_cnt',
       'popu_u20']

- q2  
    - alpha = 2.104  
    - 상위 30개  
        ['차대차', '가해_남', 'public_y', 'traffic_x', 'single_f_2030', '가해_30대',
       'report_sx_여성', 'pol_num', 'single_o60', '가해_10대이하', '차대사람', 'popu_u20',
       '피해_80대이상', '피해_30대', 'single_f_u20', '피해_50대', '가해_70대', 'crm',
       '피해_40대', '가해_20대', '가해_여', 'single_4050', 'for_2030', 'single_u20',
       '가해_40대', 'car_cnt', 'for_u20', '피해_70대', 'popu_o60', 'vio_cnt']

- q3  
    - alpha = 9.999  
    - 상위 30개  
        ['robber_murder_y', '피해_40대', 'mur_rob_cnt', '가해_80대이상', '피해_50대',
       'single_합계', 'light_cnt', 'public_y', 'robber_murder_x', 'crm',
       'popu_4050', 'traffic_y', 'cctv', '피해_20대', 'single_f_4050', '차대사람',
       'single_o60', 'single_f_2030', '가해_여', '가해_70대', '가해_40대', 'traffic_x',
       'for_2030', 'for_u20', '가해_20대', '피해_10대이하', 'pol_num', 'single_4050',
       '피해_70대', '가해_10대이하']

- q4  
    - alpha = 5.411  
    - 상위 30개  
        ['robber_murder_x', 'popu_2030', '피해_20대', 'theft_violence_y',
       '가해_성별_기타불명', 'crm', 'single_f_4050', 'single_f_합계', 'popu_o60',
       '피해_80대이상', 'vio_cnt', 'theft_violence_x', 'cctv', '5m_crm_yn',
       '피해_50대', '피해_미분류', 'traffic_y', 'for_u20', 'popu_u20', '피해_10대이하',
       'single_f_2030', 'mur_rob_cnt', '가해_70대', '가해_20대', '가해_40대',
       '가해_10대이하', 'single_4050', 'car_cnt', '피해_70대', 'pol_num']

- q5  
    - alpha = 5.517  
    - 상위 30개  
        ['report_sx_남성', 'single_f_u20', 'single_f_합계', 'report_sx_불상', 'for_합계',
       '가해_성별_기타불명', 'robber_murder_x', '가해_여', 'traffic_y', '피해_50대',
       'single_u20', 'crm', '피해_80대이상', 'single_o60', 'popu_o60', '피해_미분류',
       'light_cnt', 'theft_violence_y', 'for_u20', 'single_f_2030', '가해_10대이하',
       '가해_40대', '가해_70대', 'for_2030', '가해_20대', 'car_cnt', 'single_4050',
       'popu_u20', 'vio_cnt', '피해_70대']